### PR TITLE
invite viewer user

### LINKED ISSUE #7712: Invite viewer user

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #7712

@xavijam can you CR frontend, please?
@gfiorav can you CR backend, please?

### CODE REVIEW

**Comment by xavijam** [comment]
:+1:

File: app/models/carto/user_creation.rb [hunk 1]
```diff
@@ -141,8 +141,8 @@ def enabled?
     cartodb_user.enable_account_token.nil? && cartodb_user.enabled
   end
 
-  def unused_invitation
-    select_valid_invitation_token(Carto::Invitation.query_with_unused_email(email).all)
+  def pertinent_invitation
+    @pertinent_invitation ||= select_valid_invitation_token(Carto::Invitation.query_with_unused_email(email).all)
```
Comment (gfiorav) [inline app/models/carto/user_creation.rb]: `user_creation.rb` is never reused right?

File: app/models/carto/invitation.rb [hunk 2]
```diff
@@ -19,20 +19,22 @@ class Invitation < ActiveRecord::Base
 
     private_class_method :new
 
-    def self.create_new(inviter_user, users_emails, welcome_text)
+    def self.create_new(inviter_user, users_emails, welcome_text, viewer)
```
Comment (gfiorav) [inline app/models/carto/invitation.rb]: Only if your nuts: a `viewer?` wrapper would be nice

File: app/views/organization_mailer/invitation.html.erb [hunk 3]
```diff
@@ -6,7 +6,7 @@
       <!-- HEADER -->
       <table width="100%" border="0" cellspacing="0" cellpadding="0" style="background-color: #ffffff; border: 1px solid #dddddd; border-bottom: 0; border-collapse: separate !important; border-top-left-radius: 4px; border-top-right-radius: 4px; width: 100% !important;">
         <tr>
-          <td align="left" style="padding: 40px 50px 0 50px; font-weight: bold; font-size: 25px; font-family: Helvetica, Arial, sans-serif; color: #333333;">You are invited to join the <%= @invitation.organization.name_to_display %> organization at CartoDB</td>
+          <td align="left" style="padding: 40px 50px 0 50px; font-weight: bold; font-size: 25px; font-family: Helvetica, Arial, sans-serif; color: #333333;">You are invited to join the <%= @invitation.organization.name_to_display %> organization at CartoDB<%= @invitation.viewer? ? ' as a viewer' : '' %></td>
```
Comment (gfiorav) [inline app/views/organization_mailer/invitation.html.erb]: `viewer?` method for invitation doesn't exist right?

**Comment by gfiorav** [comment]
Made some comments! A posible missing method `viewer?` for `Carto::Invitation` being used in a template is my only real concern.

After that, fixing the conflict and 🍏, 

📦 for the free 🌍

File: app/models/carto/user_creation.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/models/carto/user_creation.rb]: If you mean that if the same instance of a `UserCreation` is ever used for different users, no, instances are not reused. As you see in the attributes it models an specific user creation, so memoizing this is not a problem (and `pertinent_invitation` is a private method).

File: app/models/carto/invitation.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/models/carto/invitation.rb]: Not needed, as [ActiveRecord add `?`-ending methods for all methods](http://stackoverflow.com/questions/12559142/using-question-mark-character-in-rails-activerecord-column-name) (I haven't found official docs on this).

File: app/views/organization_mailer/invitation.html.erb (hunk 3 above)
Comment (juanignaciosl) [inline app/views/organization_mailer/invitation.html.erb]: Answered before :-)

**Pull request merged**